src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -2042,6 +2051,54 @@ fn suggest_disambiguator(
     }
 }
 
+/// Report a link to an item that will not have documentation generated.
+fn missing_docs_for_link(cx: &DocContext<'_>, item: &Item, destination_id: DefId, path_str: &str, dox: &str, link: &MarkdownLink) {
+    let sym;
+    let item_name = match item.name {
+        Some(name) => {
+            sym = name.as_str();
+            &*sym
+        }
+        None => "<unknown>",
+    };
+    let msg =
+        format!("documentation for `{}` links to item `{}` which will not have documentation generated", item_name, path_str);
+
+    report_diagnostic(cx, PRIVATE_INTRA_DOC_LINKS, &msg, item, dox, &link.range, |diag, sp| {
+        use rustc_middle::ty::DefIdTree;
+        use crate::clean::types::{AttributesExt, NestedAttributesExt};
+
+        if let Some(sp) = sp {
+            diag.span_label(sp, "this item is will not be documented");
```

Comment:
```suggestion
            diag.span_label(sp, "this item is private and will not be documented");
```

Reply:
"private" isn't correct, there are many reasons it might not be documented (e.g. it's a public export of a private item marked with no_inline; crate is masked; --extern-html-url wasn't passed).